http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/WsmApiClientProvider.scala

```diff
@@ -11,27 +17,32 @@ import org.http4s.Uri
  * Based on `org/broadinstitute/dsde/rawls/dataaccess/workspacemanager/WorkspaceManagerApiClientProvider.scala`
  *
  */
-trait WsmApiClientProvider {
-  def getControlledAzureResourceApi(token: String): ControlledAzureResourceApi
-  def getResourceApi(token: String): ResourceApi
+trait WsmApiClientProvider[F[_]] {
+  def getControlledAzureResourceApi(token: String)(implicit ev: Ask[F, AppContext]): F[ControlledAzureResourceApi]
+  def getResourceApi(token: String)(implicit ev: Ask[F, AppContext]): F[ResourceApi]
 }
 
-class HttpWsmClientProvider(baseWorkspaceManagerUrl: Uri) extends WsmApiClientProvider {
-  private def getApiClient: ApiClient = {
-    val client: ApiClient = new ApiClient()
-    client.setBasePath(baseWorkspaceManagerUrl.renderString)
-    client
-  }
+class HttpWsmClientProvider[F[_]](baseWorkspaceManagerUrl: Uri)(implicit F: Async[F]) extends WsmApiClientProvider[F] {
+  private def getApiClient(token: String)(implicit ev: Ask[F, AppContext]): F[ApiClient] =
+    for {
+      ctx <- ev.ask
+      client = new ApiClient() {
+        override def performAdditionalClientConfiguration(clientConfig: ClientConfig): Unit = {
+          super.performAdditionalClientConfiguration(clientConfig)
+          ctx.span.foreach { span =>
+            clientConfig.register(Tracing.getTracer.withSpan(span))
```

Comment:
Pass a `Span` to the WSM call to enable cross-service tracing. Inspired by [Rawls](https://github.com/broadinstitute/rawls/blob/develop/core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/workspacemanager/WorkspaceManagerApiClientProvider.scala#L44). TODO add demo/screenshot

Reply:
Here's a span for Leo in a BEE!

Unfortunately I can't see WSM in the trace because
https://github.com/broadinstitute/terra-helmfile/blob/master/values/app/workspacemanager/bee.yaml.gotmpl#L87
```
# WSM doesn't currently have permissions for this in QA, it just spams errors
samplingProbability: 0
```

<img width="1537" alt="image" src="https://github.com/DataBiosphere/leonardo/assets/5368863/eb2dd181-2fa9-423d-8bce-7467bc106706">

Reply:
I am curious that this does not need the `JerseyTracingFilter` like rawls uses

Reply:
Um, maybe it does. Leo doesn't have the `bio.terra.common` package. Will look… 👁️

Reply:
Ah yeah of course that's needed. I added TCL dependency to Leo and added `JerseyTracingFilter` and I see WSM spans now:
<img width="1534" alt="image" src="https://github.com/DataBiosphere/leonardo/assets/5368863/c476e88c-da40-4eea-be66-16c1ef0937bb">